pkg/tsdb/tempo/tempo.go

```diff
@@ -123,6 +150,118 @@ func (s *Service) getDSInfo(ctx context.Context, pluginCtx backend.PluginContext
 	return instance, nil
 }
 
+func (s *Service) CallResource(ctx context.Context, req *backend.CallResourceRequest, sender backend.CallResourceResponseSender) error {
+	return s.resourceHandler.CallResource(ctx, req, sender)
+}
+
+// handleTags handles requests to /tags resource
+func (s *Service) handleTags(rw http.ResponseWriter, req *http.Request) {
+	s.proxyToTempo(rw, req, "api/v2/search/tags")
+}
+
+// handleTagValues handles requests to /tag-values resource
+func (s *Service) handleTagValues(rw http.ResponseWriter, req *http.Request) {
+	// Extract the encoded tag from query parameters
+	encodedTag := req.URL.Query().Get("tag")
+	if encodedTag == "" {
+		http.Error(rw, "Missing required 'tag' parameter", http.StatusBadRequest)
+		return
+	}
+
+	tempoPath := fmt.Sprintf("api/v2/search/tag/%s/values", encodedTag)
+	s.proxyToTempo(rw, req, tempoPath)
+}
+
+// proxyToTempo is the shared function that builds the URL and proxies requests to Tempo
+func (s *Service) proxyToTempo(rw http.ResponseWriter, req *http.Request, tempoPath string) {
+	ctx := req.Context()
+	pCtx := backend.PluginConfigFromContext(ctx)
+
+	// Get datasource info
+	dsInfo, err := s.getDSInfo(ctx, pCtx)
+	if err != nil {
+		s.logger.Error("Failed to get data source info", "error", err)
+		http.Error(rw, "Failed to get datasource configuration", http.StatusInternalServerError)
```

Comment:
We should stick to the format `data source` everywhere.
```suggestion
		http.Error(rw, "Failed to get data source configuration", http.StatusInternalServerError)
```